web-app/src/test/com.isa.aem/calculatorMethod/CurrencyExchangeRateRepositoryTest.java

```
@@ -0,0 +1,212 @@
+        // arrange
+        String currency = "AA";
+        // act
+        currencyExchangeRateRepository.getMinDateForSelectedCurrency(currency);
+
+    }
+
+    @Test
+    @DisplayName("Should return length>0 if currency is in file")
+
+    public void returnsNotEmptyListWhenCurrencyIsInTheFile() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "pln";
+        // act
+        Boolean result = currencyExchangeRateRepository.getRangeOfSelectedCurrency(currency).length() > 0;
+        // assert
+        assertEquals(true, result);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    @DisplayName("Should return \"No value present\" if currency not in file")
+
+    public void returnsThrowWhenCurrencyIsNotInTheFileForMethodGetRangeOfSelectedCurrency() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "aa";
+        // act
+        currencyExchangeRateRepository.getRangeOfSelectedCurrency(currency);
+    }
+
+    @Test
+    @DisplayName("Should return exchange value if currencyWant and CurrencyHave is not empty and is in File")
+
+    public void returnsExchangeValueWhenCurrencyAndDateIsCorrect() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currencyHave = "EUR";
+        String currencyWant = "PLN";
+        LocalDate dateMax = LocalDate.of(2018, 07, 27);
+        // act
+        BigDecimal result = currencyExchangeRateRepository.getExchangeValue(currencyHave, currencyWant, dateMax);
+        // assert
+        assertEquals(BigDecimal.valueOf(4.2945), result);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    @DisplayName("Should return \"No value present\" if currencyWant is not in file")
+
+    public void returnsThrowsWhenCurrencyWantIsNotInTheFile() {
```

nazwa tej metody jest niespojna z DisplayName. z kodu wyglada na to ze oczekujesz exception a nie messsage "No value present"

---

ok. zmien tylko to co sie tyczy 'return throw'
napisalas, powinien "zwrocic rzucic kiedy cos tam"...sprobuj jedno z ponizszych: 

should throw an exception jakis tam kiedy cos tam
should return an exception jakis tam kiedy cos tam (czyli zwrocic wyjatek)

---

poprawione